jetson/power_logging/power_research/inference_cycles/utils.py

```diff
@@ -0,0 +1,124 @@
+"""Utility functions for inspecting power profiles."""
+
+from collections import defaultdict
+from datetime import datetime, timedelta
+
+import matplotlib.pyplot as plt
+from tqdm import tqdm
+
+
+def parse_power_log(log_values: list[str]) -> list[tuple]:
+    """Convert each power log entry to (datetime, power).
+
+    Args:
+        log_values: List of timestamp, current and voltage as string.
+
+    Returns:
+        Processed log for easy accessing the timestamp and power.
+    """
+    result = []
+    for entry in tqdm(log_values):
+        parts = entry.strip().split(",")
+        timestamp = datetime.strptime(parts[0], "%Y%m%d-%H:%M:%S.%f")
+        voltage = float(parts[1])
+        current = float(parts[2])
+        result.append((timestamp, voltage * current))
+    return result
+
+
+def parse_latencies(
+    trt_layer_latency: dict[str, list[list[float, str]]],
+) -> defaultdict:
+    """Calculate start and end time for each layer.
+
+    Args:
+        trt_layer_latency: Dictionary containing latency data for each layer.
+
+    Returns:
+        Dictionary of cytcles where each cycle is
+        a list of tuples (cycle, start_time, end_time, duration, layer_name).
+    """
+    latency_data = defaultdict(list)
+    for layer_name, layer_times in tqdm(trt_layer_latency.items()):
+        for cycle, (execution_duration, execution_end_time) in enumerate(layer_times):
+            end_timestamp = datetime.strptime(execution_end_time, "%Y%m%d-%H:%M:%S.%f")
+            duration = timedelta(milliseconds=execution_duration)
+            start_timestamp = end_timestamp - duration
+            latency_data[cycle].append(
+                (
+                    cycle,
+                    start_timestamp,
+                    end_timestamp,
+                    execution_duration,
+                    layer_name,
+                )
+            )
+    return latency_data
```

Comment:
Are these two functions worth having in the main code base? Looks like it can be useful for some future measurement collections experiments. 

So I'd either have them as a separate PR, or make a mental note that this code exists for the future